internal/logic/cross_chain/messenger_cross_chain.go

```diff
@@ -202,10 +160,12 @@ func (c *LogicCrossChain) checkETH(ctx context.Context, layer types.LayerType, s
 	c.computeBlockBalance(ctx, layer, messages, startBalance)
 }
 
-func (c *LogicCrossChain) checkBlockBalanceOneByOne(ctx context.Context, client *ethclient.Client, messengerAddr common.Address, layer types.LayerType, messages []orm.MessageMatch) {
+func (c *LogicMessengerCrossChain) checkBlockBalanceOneByOne(ctx context.Context, client *ethclient.Client, messengerAddr common.Address, layer types.LayerType, messages []orm.MessengerMessageMatch) {
 	var startBalance *big.Int
 	var startIndex int
 	for idx, message := range messages {
+		c.checker.MessengerCrossChainCheck(layer, message)
```

Comment:
no need to check this because the cross-chain eth value is only collected from SentMessage events (RelayedMessage only has message hash).